barustenberg/src/plonk/proof_system/commitment_scheme.rs

```diff
@@ -598,76 +618,98 @@ mod tests {
         //
         // Note that each polynomial F_{k, j} \in F^{n}
         //
-        size_t n = 64;
-        size_t m = 4;
-        polynomial coeffs(n * m * t);
-        for (size_t k = 0; k < t; ++k) {
-            for (size_t j = 0; j < m; ++j) {
-                for (size_t i = 0; i < n; ++i) {
-                    coeffs[k * (m * n) + j * n + i] = fr::random_element();
+        let n = 64;
+        let m = 4;
+        let poly = Arc::new(RwLock::new(Polynomial::new(n * m * t)));
+        for k in 0..t {
+            for j in 0..m {
```

Comment:
might this be faster by starting with a mutable array and then casting it at the end?

Reply:
Fixed